src/Models/HydrostaticFreeSurfaceModels/z_star_vertical_spacing.jl

```diff
@@ -0,0 +1,213 @@
+
+#####
+##### ZStar coordinate and associated grid types
+#####
+
+""" free-surface following vertical coordinate """
+struct ZStar end
+
+"""
+    struct ZStarSpacing{R, S} <: AbstractVerticalSpacing{R}
+
+A vertical spacing for the hydrostatic free surface model that follows the free surface.
+The vertical spacing is defined by a reference spacing `Δr` and a scaling `s` that obeys
+```math
+s = (η + H) / H
+```
+where ``η`` is the free surface height and ``H`` the vertical depth of the water column
+
+# Fields
+- `Δr`: reference vertical spacing with `η = 0`
+- `sⁿ`: scaling of the vertical coordinate at time step `n`
```

Comment:
Are there any options other than `s`? It doesn't produce the most obvious expressions and also `s` is used for salinity.